Khan/perseus
# Update store for `range` and `snapStep` when props change

## Summary:
Update reducer state for `range` and `snapStep` when props change (for editing experience).

https://github.com/Khan/perseus/assets/16308368/419f4933-d888-4160-9bfa-3b0a72f1eb0d

Issue: [LEMS-1824](https://khanacademy.atlassian.net/browse/LEMS-1824)

## Test plan:
- Go to the "With Locked Points" story
  - `?path=/story/perseus-editor-widgets-interactive-graph-editor--with-locked-points`
- Move the points around to see the step they move by
- Change `snapStep` to 5, 5
- Note the snap now works on 5, 5

[LEMS-1824]: https://khanacademy.atlassian.net/browse/LEMS-1824?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
